patches/api/0346-Flying-Fall-Damage-API.patch

```diff
@@ -0,0 +1,33 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: TreyRuffy <TreyRuffy@users.noreply.github.com>
+Date: Sat, 21 Aug 2021 16:02:50 -0600
+Subject: [PATCH] Flying Fall Damage API
+
+
+diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
+index f5ded992672dde910cd4e1be50c238b142157ff8..21e87d53426f7e74dd1e537e9559098445a531a1 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -1285,6 +1285,22 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+      */
+     public void setAllowFlight(boolean flight);
+ 
++    // Paper start - flying fall damage
++    /**
++     * Allows you to enable fall damage while {@link #getAllowFlight()} is {@code true}
++     *
++     * @param flyingFallDamage Enables fall damage when {@link #getAllowFlight()} is true
++     */
++    public void setFlyingFallDamage(boolean flyingFallDamage);
++
++    /**
++     * Allows you to get if fall damage is enabled while {@link #getAllowFlight()} is {@code true}
++     *
++     * @return True if fall damage is enabled when {@link #getAllowFlight()} is true
++     */
++    public boolean hasFlyingFallDamage();
++    // Paper end
```

Comment:
Any reason why this doesn't use a tristate, where it "overrides" the default behavior of the config option? This way it could also save only if it isn't UNSET.